### PR TITLE
add testgrid_publish test definition

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
Add testgrid test definition

**Release note**:

```
Publish k8s e2e conformance test results to testgrid
```

### CODE REVIEW

File: .test-defs/Testgrid.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+spec:
+  owner: DL_5C5BE3E2970B9F404D0E2F50@sap.com
+  recipientsOnFailure:
+  - DL_5C5BE3E2970B9F404D0E2F50@sap.com
+
+  description: Run kubernetes conformance tests and push result files (e2e.log and junit_01.xml) to testgrid repository.
+
+  activeDeadlineSeconds: 10800
+
+  config:
```
Comment (schrodit) [inline .test-defs/Testgrid.yaml]: Please remove indent to be consistent with `recipientsOnFailure`
```suggestion
  config:
  - type: env
     name: PUBLISH_TO_TESTGRID
     value: "true"
```

File: test/k8s-e2e/e2e-test-execute [hunk 2]
```diff
@@ -67,7 +67,8 @@ if [[ "$TESTDESCRIPTION_NAME" == "conformance.desc" ]]; then
 	conformance_test=$TRUE
 	echo -e "$(info INFO:) runnning Conformance e2e tests"
 fi
-if [[ "$TESTDESCRIPTION_NAME" =~ ^[[:digit:]]\.[[:digit:]]{1,2}$ ]]; then
+set +u
+if [ "$PUBLISH_TO_TESTGRID" = "true" ]; then
```
Comment (schrodit) [inline test/k8s-e2e/e2e-test-execute]: Maybe use same `true` as for other tests?

Comment (OlegLoewen) [reply test/k8s-e2e/e2e-test-execute]: Done

File: .test-defs/e2eSlow.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  name: e2e-slow
+spec:
+  owner: DL_5C5BE3E2970B9F404D0E2F50@sap.com
+  recipientsOnFailure:
+  - DL_5C5BE3E2970B9F404D0E2F50@sap.com
+
+  description: Run kubernetes e2e test with tags Slow and Serial.
+
+  activeDeadlineSeconds: 10800
+  labels: ["default"]
```
Comment (schrodit) [inline .test-defs/e2eSlow.yaml]: Also add the slow label, so we can easily run slow tests in the daily testrun

File: test/k8s-e2e/descriptions/blacklist.desc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+[sig-scalability] Load capacity [Feature:ManualPerformance] should be able to handle 30 pods per node ReplicationController with 0 secrets, 0 configmaps and 2 daemons
+[sig-scalability] Load capacity [Feature:Performance] should be able to handle 30 pods per node ReplicationController with 0 secrets, 0 configmaps and 0 daemons
+[sig-scheduling] SchedulerPredicates [Serial] validates local ephemeral storage resource limits of pods that are allowed to run [Feature:LocalStorageCapacityIsolation]
+[sig-network] NoSNAT [Feature:NoSNAT] [Slow] Should be able to send traffic between Pods without SNAT
+[sig-storage] In-tree Volumes [Driver: iscsi][Feature:Volumes] [Testpattern: Inline-volume (default fs)] subPath should be able to unmount after the subpath directory is deleted
+[sig-scheduling] ResourceQuota [Feature:Initializers] should create a ResourceQuota and capture the life of an uninitialized pod.
+[sig-network] [Feature:PerformanceDNS][Serial] Should answer DNS query for maximum number of services per cluster
```
Comment (schrodit) [inline test/k8s-e2e/descriptions/blacklist.desc]: This is only blacklisted until the etccd problems are fixed?

File: test/k8s-e2e/e2e-test-execute [hunk 5]
```diff
@@ -171,45 +207,72 @@ function build_k8s() {
...
+			grep "$(escape_test_name "${e2e_test}")" $all_tests_desc_path >> "$generated_run_desc_path"
 		done <<<"$(cat "${test_desc_path}")"
-	fi
 
-	if [[ $SERIAL_TESTS_ONLY -eq $TRUE ]]; then
-		sed -i '/\[Serial\]/!d' "$generated_run_desc_path"
-	elif [[ $SERIAL_TESTS_ONLY -eq $FALSE ]]; then
-		sed -i '/\[Serial\]/d' "$generated_run_desc_path"
+		# remove duplicates
+		sort -r -u -o "$generated_run_desc_path" "$generated_run_desc_path"
```
Comment (schrodit) [inline test/k8s-e2e/e2e-test-execute]: why can there be duplicates?

Comment (OlegLoewen) [reply test/k8s-e2e/e2e-test-execute]: Sometimes testcase names only differ in suffix parts. E.g. testcase1=foo and testcase2=foobar. In this case matching for one test case may result in multiple matches if we look for testcases foo and foobar.

File: test/k8s-e2e/descriptions/blacklist.desc (hunk 4 above)
Comment (OlegLoewen) [reply test/k8s-e2e/descriptions/blacklist.desc]: Yes, depending on decision of etcd and PO. Eventually we say, that we cannot support this for current time being.

File: .test-defs/e2eSlow.yaml (hunk 3 above)
Comment (OlegLoewen) [reply .test-defs/e2eSlow.yaml]: Replaced default with slow

File: .test-defs/e2eFast.yaml [hunk 6]
```diff
@@ -14,23 +14,25 @@
...
   activeDeadlineSeconds: 10800
   labels: ["default"]
-  behavior: ["serial"]
 
   config:
   - type: env
-    name: SERIAL_TESTS_ONLY
-    value: "0" # 0 = true
+    name: CONFORMANCE_SKIP
+    value: "Serial\|Slow"
```
Comment (schrodit) [inline .test-defs/e2eFast.yaml]: Please fix escaping

**Comment by OlegLoewen** [comment]
Found a bug in execute script, which needs to be fixed first.

**Pull request merged**